### PR TITLE
Check rbtree before looking for ARP cache slot

### PR DESCRIPTION
**Pull request by quantabase13**
Check whether the RB tree contains the input ip's corresponding entry first before go into the for-loop finding available ARP cache slot.

### CODE REVIEW

**Review by jserv — CHANGES_REQUESTED** [review]
Read this article [How to Write a Git Commit Message](https://cbea.ms/git-commit/) and refine the subject of git commit message(s).

File: src/arp.c [hunk 1]
```diff
@@ -70,6 +70,18 @@ int arp_cache_insert(in_addr_t ip_addr,
     if (ip_addr == 0)
         return 0;
 
+    struct arp_cache_entry find = {
+        .ip_addr = ip_addr,
+    };
+
+    if ((entry = RB_FIND(arp_cache_tree, &arp_cache_head, &find)) >0){
+        if (memcmp(entry->haddr, haddr, sizeof(mac_addr_t)) != 0){
```
Comment (jserv) [inline src/arp.c:78]: It looks weird. Can you simplify?

Comment (quantabase13) [reply src/arp.c:78]: Yes, I ' ll try to simplify the code. By the way, shall I convert the PR state to draft first?

Comment (jserv) [reply src/arp.c:75]: Keep the same pull request for reviewing.

File: src/arp.c [hunk 2]
```diff
@@ -70,6 +71,14 @@ int arp_cache_insert(in_addr_t ip_addr,
     if (ip_addr == 0)
         return 0;
 
+    if ((entry = arp_cache_get_entry(ip_addr))>0) {
```
Comment (jserv) [inline src/arp.c:74]: Run `clang-format -i` to ensure the consistent style.

Comment (quantabase13) [reply src/arp.c]: I ensured the consistent style.

File: src/arp.c [hunk 3]
```diff
@@ -70,6 +71,14 @@ int arp_cache_insert(in_addr_t ip_addr,
     if (ip_addr == 0)
         return 0;
 
+    if ((entry = arp_cache_get_entry(ip_addr)) > 0) {
+        if (memcmp(entry->haddr, haddr, sizeof(mac_addr_t)) != 0) {
```
Comment (jserv) [inline src/arp.c:75]: If the contents pointed by `entry->haddr` and `haddr` are the same, why would you duplicate?

Comment (quantabase13) [reply src/arp.c:75]: I duplicate `haddr` to `entry->haddr` when the return value of memcmp not equal to zero. According to `man memcmp` (if I'm not misunderstood) only when two contents are the same that the return value would be zero.

Comment (quantabase13) [reply src/arp.c:75]: Sorry for my last comment (I finds out it's a bit offensive).
Could I have more hints about where should I modify the code ? Thanks.

Comment (jserv) [reply src/arp.c:75]: You can consolidate the comments and commit messages for the proposed changes.

**Comment by jserv** [comment]
Squash the git commits and write down meaning commit messages to address your effort.

**Comment by jserv** [comment]
> Squash the git commits and write down meaning commit messages to address your effort.

@quantabase13, squash the git commits, meaning that there should be a single git commit associated with this pull request. Meanwhile, refine the commit messages.

**Pull request merged**